Automattic/woocommerce-payments
# Fix errors for apple pay registration

Fixes #4511

#### Changes proposed in this Pull Request
- when merchant is not in the Apple Pay [supported country list](https://support.apple.com/en-us/HT207957
), or `apple-developer-merchantid-domain-association` file has different permissions than 644, the apple domain registration is not called anymore
- for the conditions above, a notice is shown in the WooCommerce Payments admin panel

<img width="1733" alt="Screenshot 2022-08-09 at 14 37 20" src="https://user-images.githubusercontent.com/82826872/183638041-92a07470-717e-47f0-b1a8-5265774d169b.png">







#### Testing instructions
1. Set the merchant country other than the ones listed in https://support.apple.com/en-us/HT207957 .
2. Enable Apple Pay
3. `Apple Pay: You are not in the countries and regions that support Apple Pay. You can find a full list of countries and regions at https://support.apple.com/en-us/HT207957` should be displayed in Payment methods panel
4. Rename `docker/wordpress/.well-known/apple-developer-merchantid-domain-association` file
5. `Apple Pay: Invalid domain association file permissions. /var/www/html/.well-known needs to have 755 permissions and /var/www/html/.well-known/apple-developer-merchantid-domain-association 644. Set the proper permissions and disable/enable Apple Pay.`  should be displayed in Payment methods panel




*

-------------------


**Post merge**
